src/org/openvv/js/OVVAsset.js

```diff
@@ -872,6 +925,15 @@ function OVVAsset(uid, dependencies) {
             check.error = 'Player not found!';
             return check;
         }
+
+        if($ovv.intersectionObserverSupported){
+            check.technique = OVVCheck.INTERSECTIONOBSERVER;
+            checkIntersectionObserver(check,player);
+            check.viewabilityState = check.percentViewable >= 50 ? OVVCheck.VIEWABLE : OVVCheck.UNVIEWABLE;
+        
+            return check;
```

Comment:
IntersectionObserver does not take DOM obstruction or CSS hiding into consideration. Therefore, returning the OVVCheck object at this point will cause us to lose DOM obstruction and CSS calculations.